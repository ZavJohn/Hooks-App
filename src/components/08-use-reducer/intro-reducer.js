const initialState = [{
    id: 1,
    toDo: 'Comprar pan',
    done: false
}];

const toDoReducer = ( state = initialState, action ) => {

    if( action?.type === 'Agregar' ){
        return [ ...state, action.payLoad ];
    }
    
    return state;
}

let todos = toDoReducer();

const newToDo = {
    id: 2,
    toDo: 'comprar leche',
    done: false
}

const addToDoAction = {
    type: 'Agregar',
    payLoad: newToDo
}

todos = toDoReducer( todos, addToDoAction );




console.log(todos);