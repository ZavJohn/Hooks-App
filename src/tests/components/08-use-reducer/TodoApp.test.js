
import React from 'react';
import { mount, shallow } from 'enzyme';
import { TodoApp } from '../../../components/08-use-reducer/TodoApp';
import { demosTodos } from '../../fixtures/demosTodos';
import '@testing-library/jest-dom';
import { act } from '@testing-library/react-hooks';

describe('Pruebas del <TodoApp/>', () => {
    
    const wrapper = shallow( <TodoApp />);

    Storage.prototype.setItem = jest.fn( () => {})

    test('Debe mostrarse correctamente', () => { 

        expect( wrapper ).toMatchSnapshot();
     });

     test('Debe de agregar un Todo', () => { 

        const wrapper = mount( <TodoApp /> );

        act( () => {
            wrapper.find('TodoAdd').prop('handleAddTodo')( demosTodos[0] );
            wrapper.find('TodoAdd').prop('handleAddTodo')( demosTodos[1] );
        });
        

        expect( wrapper.find('h2').text().trim() ).toBe( 'ToDos App (2)' );
        expect( localStorage.setItem ).toHaveBeenCalledTimes(2);
     });


});