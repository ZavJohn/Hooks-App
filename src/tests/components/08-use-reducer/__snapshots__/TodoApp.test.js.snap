// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del <TodoApp/> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
     ToDos App (
    0
    ) 
  </h2>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-7"
    >
      <TodoList
        handleDelete={[Function]}
        handleToggle={[Function]}
        todos={Array []}
      />
    </div>
    <div
      className="col-5"
    >
      <TodoAdd
        handleAddTodo={[Function]}
      />
    </div>
  </div>
</Fragment>
`;
